aten/src/ATen/native/native_functions.yaml

```diff
@@ -263,8 +263,12 @@
 
 - func: squeeze(Tensor self, int64_t dim) -> Tensor
 
+- func: squeeze(Tensor self, IntList dims) -> Tensor
```

Comment:
Question: does this make the python signature use the keyword argument `dims`?

Reply:
It will use both.

```python
In [1]: import torch

In [2]: v = torch.autograd.Variable(torch.ones([2, 1, 3, 1]), requires_grad=True)

In [3]: v.squeeze(dim=1).size()
Out[3]: torch.Size([2, 3, 1])

In [4]: v.squeeze(dims=[1, 2]).size()
Out[4]: torch.Size([2, 3, 1])

In [5]: v.squeeze(dim=[1, 2]).size()
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-5-f2e3d2988f84> in <module>()
----> 1 v.squeeze(dim=[1, 2]).size()

RuntimeError: squeeze() received an invalid combination of arguments - got (dim=list, ), but expected one of:
 * ()
      didn't match because some of the keywords were incorrect: dim
 * (int dim)
      didn't match because some of the arguments have invalid types: (dim=list, )
 * (tuple of ints dims)
      didn't match because some of the keywords were incorrect: dim
```

Reply:
We should stick to `dim` as the argument name. Different names complicates the signature and documentation.

Reply:
I think there is (was?) some bug in the jit that prevented this.

Reply:
Initially, I had the same argument name. But it resulted in compilation errors like this:
  File "/home/yolo/src/pytorch/tools/jit/gen_jit_dispatch.py", line 113, in gen_jit_dispatch
    assert descriptor not in ops, descriptor
AssertionError: squeeze-1-dim

Reply:
We should patch the JIT dispatch generator to include argument type and not only name in the descriptor (it fails because we have ambiguous overloads)